FlowEquivalentForest/Flow.lean

```diff
@@ -342,9 +254,10 @@ lemma Flow.fromPath.value_eq_bottleneck
     (Flow.fromPath h P).value = G.bottleneck h P := sorry
 
 lemma flow_to_self_zero {P : FlowProblem G} (F : Flow P) (v : V) : F.f v v = 0 := by
-    have h : F.f v v = -F.f v v := F.skewSymmetry
-    linarith
+    apply Or.elim (@Flow.skewSymmetry _ _ _ _ F v v)
```

Comment:
While we are at it: I didn't misunderstood what `\forall {u, v}` does when writing it and from this example it is immediate that I used it wrong :D I now think that none of the definitions in `Flow` should use curly braces and `u` and `v` should be passed explicitly. What do you think?

(Using `\forall {u, v}` is like `def foobar {u : V} := ...`, i.e. `u` is passed implicitly. Like with `def`, `theorem`, etc., it doesn't really make sense to use in `\forall` unless there is a later argument that would help with inference of `u`; see for example in `PairMatrix` (definition: `{x y : α} → (hxy : x ≠ y) → β`) where it absolutely makes sense, because `x` and `y` can be completely inferred form the hypothesis `hxy`)

Reply:
Yesn't, it could make sense in some scenarios, using just `foobar` fills any goal of the form `foobar v`. To my knowledge the mathlib sometimes uses definitions of this form even when arguments cannot be inferred.
There should be a better solution for the line you quoted, this was just my way to make it work because I didn't know anything better. But you could theoretically work with `apply Or.elim F.skewSymmetry`, it looks nice but potentially makes the next step more tedious.

So the easy fix would be to only use implicit arguments when they can be inferred by other arguments. I don't know what the best solution is, but the easy fix sounds practical to me.